### PR TITLE
Add support fopen

### PR DESCRIPTION
**Pull request by sava-cska**
## Checklist:
- [ ] The PR addresses a single issue.  If it can be divided into multiple independent PRs, please do so.
- [ ] The PR is divided into a logical sequence of commits OR a single commit is sufficient.
- [ ] There are no unnecessary commits (e.g. commits fixing issues in a previous commit in the same PR).
- [ ] Each commit has a meaningful message documenting what it does.
- [ ] All messages added to the codebase, all comments, as well as commit messages are spellchecked.
- [ ] The code is commented OR not applicable/necessary.
- [ ] The patch is formatted via clang-format OR not applicable (if explicitly overridden leave unchecked and explain).
- [ ] There are test cases for the code you added or modified OR no such test cases are required.

### CODE REVIEW

File: runtime/POSIX/input_output.c [hunk 1]
```diff
@@ -1,7 +1,82 @@
...
+#include <stdlib.h>
+#include <fcntl.h>
+#include <errno.h>
+
+/*
+ * Return the (stdio) flags for a given mode. Store the flags
+ * to be passed to an open() syscall through *optr.
+ * Return 0 on error.
+ */
+int count_flags(const char *mode, int *optr) {
```
Comment (alexey-utkin) [inline runtime/POSIX/input_output.c:11]: Bad naming. Nothing is counting, but `convert_to_stdio_open_flags`.
For such a namning you don't need a function doc)

File: runtime/POSIX/input_output.c [hunk 2]
```diff
@@ -1,7 +1,82 @@
...
+  } else {
+    if ((f = open(file, oflags)) < 0)
+      return NULL;
+  }
+  fp->_fileno = f;
+  fp->_mode = oflags;
+  return fp;
+}
+
+int file_descriptor(FILE *stream) {
```
Comment (alexey-utkin) [inline runtime/POSIX/input_output.c:61]: `get_file_descriptor`?

File: runtime/POSIX/input_output.c (hunk 1 above)
Comment (sava-cska) [reply runtime/POSIX/input_output.c:11]: Change

File: runtime/POSIX/input_output.c (hunk 2 above)
Comment (sava-cska) [reply runtime/POSIX/input_output.c]: Change

**Review by alexey-utkin — APPROVED** [review]

**Review by ladisgin — APPROVED** [review]

**Pull request merged**